pyomo/contrib/doe/doe.py

```diff
@@ -412,240 +409,99 @@ def _sequential_finite(self, read_output, extract_single_model, store_output):
-                    mod = self.discretize_model(mod)
-
-                # solve model
-                time0_solve = time.time()
-                square_result = self._solve_doe(mod, fix=True)
-                time1_solve = time.time()
-                time_allsolve.append(time1_solve-time0_solve)
-                models.append(mod)
-
-                if extract_single_model:
-                    mod_name = store_output + str(no_s) + '.csv'
-                    dataframe = extract_single_model(mod, square_result)
-                    dataframe.to_csv(mod_name)
+            mod = pyo.ConcreteModel()
 
-                # loop over measurement item and time to store model measurements
-                output_iter = []
+            mod.scena = pyo.Set(initialize=list(range(len(self.scenario_list))))
 
-                for j in self.flatten_measure_name:
-                    for t in self.flatten_measure_timeset[j]:
-                        measure_string_name = self.measure.SP_measure_name(j,t,mode='sequential_finite')
-                        C_value = pyo.value(eval(measure_string_name))
-                        output_iter.append(C_value)
+            # Allow user to self-define complex design variables
+            #mod.t0 = pyo.Set(initialize=[0])
+            #mod.t_con = pyo.Set(initialize=[0, 0.125, 0.25, 0.375, 0.5, 0.625, 0.75, 0.875, 1])
+            #mod.CA0 = pyo.Var(mod.t0, bounds=[1,5], within=pyo.NonNegativeReals)
+            #mod.T = pyo.Var(mod.t_con, bounds=[300, 700], within=pyo.NonNegativeReals)
 
-                output_record[no_s] = output_iter
+            mod.add_component('t0', pyo.Set(initialize=[0]))
+            mod.add_component('t_con', pyo.Set(initialize=[0, 0.125, 0.25, 0.375, 0.5, 0.625, 0.75, 0.875, 1]))
+            mod.add_component('CA0', pyo.Var(mod.t0, bounds=[1,5], within=pyo.NonNegativeReals))
+            mod.add_component('T', pyo.Var(mod.t_con, bounds=[300, 700], within=pyo.NonNegativeReals))
 
-            output_record['design'] = self.design_values
-            if store_output:
-                f = open(store_output, 'wb')
-                pickle.dump(output_record, f)
-                f.close()
+            def block_build(b,s):
+                self.create_model(m=b)
+                
+                for par in self.param:
+                    par_strname = eval('b.'+str(par))
+                    par_strname.fix(scena_gen["scenario"][s][par])
 
-            # calculate jacobian
-            jac = self._finite_calculation(output_record, scena_gen)
-
-            time11 = time.time()
-            self.logger.info('Build time with sequential_finite mode [s]:  %s', sum(time_allbuild))
-            self.logger.info('Solve time with sequential_finite mode [s]:  %s', sum(time_allsolve))
-            self.logger.info('Total wall clock time [s]:  %s', time11-time00)
-
-            # return all models formed
-            self.models = models
+            mod.lsb = pyo.Block(mod.scena, rule=block_build)
```

Comment:
What does `lsb` stand for? Can we think of a more descriptive variable name? If not, please add a comment explaining what this is doing (and define 'lsb').

Reply:
I change `lsb` to `block` now.